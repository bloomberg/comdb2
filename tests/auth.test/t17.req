# OP user
set user 'user1'
set password 'password1'
select '---------- OP user ----------' as user;
# create an anonymous user
put password '' for ''

# create a table and grant read to the anonymous user
create table t1(i int)$$
insert into t1 values(1);
grant read on t1 to ''

set user ''
set password ''
select '---------- anonymous user ----------' as user;
select i=1 from t1;
# the following commands must fail
insert into t1 values(2);
drop table t1;

# Cleanup
set user 'user1'
set password 'password1'
select '---------- OP user ----------' as user;
drop table t1;
