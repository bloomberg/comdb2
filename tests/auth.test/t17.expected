(user='---------- OP user ----------')
(rows inserted=1)
(user='---------- anonymous user ----------')
(i=1=1)
[insert into t1 values(2)] failed with rc -106 Write access denied to t1 for user  bdberr=15
[drop table t1] failed with rc -3 User does not have OP credentials
(user='---------- OP user ----------')
